src/org/omegat/core/segmentation/SRX.java

```diff
@@ -379,11 +438,38 @@ public static SRX getDefault() {
             MapRule maprule = getMappingRules().get(i);
             if (maprule.getCompiledPattern().matcher(srclang.getLanguage()).matches()) {
                 rules.addAll(maprule.getRules());
+                if (! this.cascade) {
```

Comment:
`!` should not have trailing whitespace.

Reply:
Is this sufficient to implement support for cascading? If so, why have a getter for the `cascade` property? Also I'd really like to see the expected behavior nailed down in some unit tests.

Reply:
> Is this sufficient to implement support for cascading?

If we look to the definition in SRX specification : " If the cascade attribute is set to "no" then aggregation of the <rule> elements will terminate. ". That is what I do here.

>  If so, why have a getter for the cascade property? 

The good question is : why do we have getters in this class, after all? Why do we have getters for segment subflows, which are not supported?
Don't forget one thing: I did not write this class, I just modified it. I added this getter because other parameters, even if not really used, did have one, so I tried to remain coherent.

Reply:
> Also I'd really like to see the expected behavior nailed down in some unit tests.

That's not so easy.
Writing a unit test will only prove that the segmenter works as I wrote it. But a correct test would be to have a normative sample, with expected behaviour, and check that in both cases (with cascade = true then with cascade = false) the segmenter gives the correct result.
Do you have such an example?

Reply:
> The good question is : why do we have getters in this class, after all? Why do we have getters for segment subflows, which are not supported?
> Don't forget one thing: I did not write this class, I just modified it. I added this getter because other parameters, even if not really used, did have one, so I tried to remain coherent.

I haven't forgotten that you didn't write the class, and I am not blaming you for code written by those who came before you. I simply wanted to understand why you wrote it the way you did; it appears the answer is "to remain coherent". That's a perfectly reasonable answer.

Now that I understand your reason, we can think about whether or not coherency is a good thing in this instance, given that the current code has some obvious problems. I would be OK with you not adding this new getter, but I don't feel that strongly about it.

Reply:
>  Writing a unit test will only prove that the segmenter works as I wrote it.

This, in and of itself, has at least some value.

> But a correct test would be to have a normative sample, with expected behaviour, and check that in both cases (with cascade = true then with cascade = false) the segmenter gives the correct result.
> Do you have such an example?

No, but as the author of this feature I figured you would.

Reply:
> Now that I understand your reason, we can think about whether or not coherency is a good thing in this instance, given that the current code has some obvious problems. I would be OK with you not adding this new getter, but I don't feel that strongly about it.

In the meantime I found a location where these accessors may/should be used.
Actually, your segmentation customizer (GUI) does not allow to modify these parameters. However, you have a class org.omegat.core.segmentation.datamodels.SRXOptionsModel which seems written for that but actually unused (I checked that it is called nowhere)
This is the only class which uses getters and setters for segment subflows (but actually it does not support cascade)

Now the questions are:
1. do you plan at any time to implement a window or a panel to edit these fields? if not, shouldn't we delete SRXOptionsModel?
2. Now that cascade is really supported by OmegaT, shouldn't we add a checkbox in the already existing GUI panel?
3. More complicated. What to do when the user checks/unchecks the box?
Two possibilities.

a. The simplest, of course, is to simply set the boolean value.
But if you know what cascade means, you certainly understand that all language rules are affected when you uncheck the box : all rules coming from next language maps are not used anymore.
So, second possibility:
b. In my project, for which I just sent you the link in previous message, I also implemented the possibility to "uncascade" an SRX file: not only it sets cascade="no", but it copies all the rules which would have applied to a given language in the correct location, ensuring that the resulting file is equivalent to the previously cascaded file.

My proposal would be the following: add a checkbox in the GUI, but when the user unchecks it, display a dialog "do you want to uncascade current rules?", with 3 possibilities: 1) YES (apply the algorithm and refreshes the editor), 2) NO (uncheck the box but do not touch the rules), 3) CANCEL (leave all as is, keep the box checked)

What do you think?